ui/components/src/Legend/Legend.tsx

```diff
@@ -30,7 +44,50 @@ export interface LegendProps {
 // future as people test this out on different machines.
 const NEED_VIRTUALIZATION_LIMIT = 500;
 
-export function Legend({ width, height, options, data }: LegendProps) {
+export function Legend({ width, height, options, data, selectedItems, onSelectedItemsChange }: LegendProps) {
+  const onLegendItemClick = (e: React.MouseEvent<HTMLElement, MouseEvent>, seriesId: string) => {
+    const isModifiedClick = e.metaKey || e.shiftKey;
+
+    const newSelected = produce(selectedItems, (draft) => {
+      if (draft === 'ALL') {
+        return {
+          [seriesId]: true,
+        };
+      }
+
+      const isSelected = !!draft[seriesId];
+
+      // Clicks with modifier key can select multiple items.
+      if (isModifiedClick) {
+        if (isSelected) {
+          // Modified click on already selected item. Remove that item.
+          delete draft[seriesId];
+        } else {
+          // Modified click on not-selected item. Add it.
+          draft[seriesId] = true;
+        }
+        return draft;
+      }
+
+      if (isSelected) {
+        // Clicked item was already selected. Unselect it and return to
+        // ALL state.
+        return 'ALL' as const;
```

Comment:
instead of setting it to 'ALL' state, do you think it should only deselect the one that you click?

Reply:
I think that would make more sense with a checkbox interface, but that's not what the list legend has right now. You can unselect a single item with a modified click (the conditional above). I think we're going to consider moving this to a checkbox interface after we get the table legend work (which does use checkboxes) working, so this logic will probably change in the near future. 

In the near term, the goal with this PR is to keep the logic as it was, but to move it down more closely to the legend component (instead of having it live in time series panel).